File: src/mb/mizinkobusters/kitpvp/kit/Astronaut.java
package mb.mizinkobusters.kitpvp.kit;

import mb.mizinkobusters.kitpvp.utils.KitPvPUtils;
import mb.mizinkobusters.kitpvp.utils.PlayerSalvationUtils;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.event.player.PlayerRespawnEvent;

public class Astronaut implements Listener {

    @EventHandler
    public void onKill(PlayerRespawnEvent event) {
        Player player = event.getPlayer();
        if (player.getKiller() == null) {
            return;
        }

        Player killer = player.getKiller();
        if (!KitPvPUtils.isInWorld(killer)) {
            return;
        }
        if (!KitPvPUtils.getKit(killer).equals("Astronaut")) {
            return;
        }
        PlayerSalvationUtils.heal(player);
    }

    @EventHandler
    public void onDamage(EntityDamageEvent event) {
        if (!event.getEntity().getType().equals(EntityType.PLAYER)) {
            return;
        }

        Player player = (Player) event.getEntity();
        if (!KitPvPUtils.getKit(player).equals("Astronaut")) {
            return;
        }

        if (!event.getCause().equals(EntityDamageEvent.DamageCause.FALL)) {
            return;
        }
        event.setCancelled(true);
    }
    // TODO ダブルジャンプの実装
}

File: src/main/java/mb/mizinkobusters/kitpvp/kit/Astronaut.kt
package mb.mizinkobusters.kitpvp.kit

import org.bukkit.Material
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.event.entity.EntityDamageEvent.DamageCause
import org.bukkit.inventory.ItemStack
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

object Astronaut : BaseKit {
    override val displayItemStack = ItemStack(Material.GLASS)
    override val lore = listOf(
        "§f頭: ガラス",
        "§f胴: 鉄",
        "§f腰: 鉄",
        "§f脚: 鉄",
        "§f武器: 鉄の剣",
        "§f補助: 金のリンゴ×2",
        "§f特殊: 跳躍力上昇III",
        "§f特殊: 移動速度低下II",
        "§f特殊: 落下ダメージ無効"
    )

    override val helmet = ItemStack(Material.GLASS)
    override val chestplate = ItemStack(Material.IRON_CHESTPLATE)
    override val leggings = ItemStack(Material.IRON_LEGGINGS)
    override val boots = ItemStack(Material.IRON_BOOTS)

    override val weapons = listOf(
        ItemStack(Material.IRON_SWORD),
        ItemStack(Material.GOLDEN_APPLE, 2),
    )

    override val effects: List<PotionEffect> = listOf(
        PotionEffect(PotionEffectType.JUMP, 1000000, 2, false, false),
        PotionEffect(PotionEffectType.SLOW, 1000000, 1, false, false)
    )

    override fun onDamage(event: EntityDamageEvent) {
        super.onDamage(event)

        if (event.cause == DamageCause.FALL) {
            event.isCancelled = true
        }
    } // TODO ダブルジャンプの実装
}